libscript/src/sort.mlc

```diff
@@ -245,4 +245,66 @@ end syntax
 
 --
 
+/*
+Summary: A handler type that can be used in the <SortListUsingHandler> command.
+
+Parameters:
+pLeft: The left hand value to compare.
+pRight: The right hand value to compare.
+
+Returns: 
+An integer less than 0 if pLeft is less than pRight with respect to the intended ordering,
+an integer greater than 0 if pLeft is greater than pRight with respect to the intended ordering,
+or 0 if pLeft is equal to pRight.
+
+Description:
+Any handler of type <SortCompare> can be passed to <SortListUsingHandler> to sort a list. 
+It takes two arguments, and returns an integer based on the comparison.
+
+Tags: Sorting
+*/
+public handler type SortCompare(in pLeft as any, in pRight as any) returns Integer
+
+public foreign handler MCSortExecSortListUsingHandler(inout Target as List, in Handler as SortCompare) returns nothing binds to "<builtin>"
+
+/*
+Summary:        Sorts <Target> using <Handler> as a comparison function.
+Target:         An expression that evaluates to a list.
+Handler: 		A handler of type <SortCompare>
+
+Example:
+ handler RandomSort(in pLeft as any, in pRight as any) returns Integer
+ 	if any number > 0.5 then
+ 		return 1
+ 	else
+ 		return -1
+ 	end if
+ end handler
```

Comment:
Are you sure this is a valid comparison function? If I remember correctly (I might not), a comparison function for stable sort has to impose a total ordering, i.e. if A > B and B > C then A > C.  With the random result, you could have A > B, B > C and C < A.

What about adding recommendation to us a consistent comparison function in the docs? You could change this example to sorting a list of lists on the Nth element of each sublist, e.g. something vaguely like:

````
variable sKeyIndex as Number

private handler CompareListsByElement(in pLeft as any, in pRight as any) returns Integer
   variable tLeft as Number
   variable tRight as Number
   put pLeft[sKeyIndex] into tLeft
   put pRight[sKeyIndex] into tRight

   if tLeft > tRight then
      return 1
   else if tLeft < tRight then
      return -1
   else
      return 0
   end if
end handler

public handler TestSortListOfLists(in pList as List, in pKeyIndex as Number) returns List
   put pKeyIndex into sKeyIndex
   sort pList using handler CompareListsByElement
   return pList
end handler
````